runtime/Dart/lib/src/prediction_context.dart

```diff
@@ -403,22 +418,32 @@ abstract class PredictionContext {
       }
     }
 
+    var safeMergedReturnStates = <int>[];
+    var safeMergedParents = <PredictionContext>[];
+
     // trim merged if we combined a few that had same stack tops
     if (k < mergedParents.length) {
       // write index < last position; trim
       if (k == 1) {
         // for just one merged element, return singleton top
         PredictionContext a_ = SingletonPredictionContext.create(
-            mergedParents[0], mergedReturnStates[0]);
+          mergedParents[0]!,
+          mergedReturnStates[0]!,
+        );
         if (mergeCache != null) mergeCache[Pair(a, b)] = a_;
         return a_;
       }
-      mergedParents = List(k)..setRange(0, k, mergedParents);
-      mergedReturnStates = List(k)..setRange(0, k, mergedReturnStates);
+      safeMergedParents = List.generate(k, (n) => mergedParents[n]!);
+      safeMergedReturnStates = List.generate(k, (n) => mergedReturnStates[n]!);
+    } else {
+        safeMergedParents = List.generate(mergedParents.length, (n) => mergedParents[n]!);
```

Comment:
I wasnt able to do `mergedParents = mergedParents as List<PredictionContext>`, because `List<PredictionContext>`  is not a subtype of `List<PredictionContext?>`... So I endedup iterating and creating a new list... seems to be a bit overkill... let me know if you have any alternatives.

Reply:
Is there a reason that mergedParent has to be not null?

Reply:
Is there a reason that mergedParents has to be not nullable list?

Reply:
yeah I guess it doesnt make sense to force that on mergedParents. 

How about mergedReturnStates? Can I default them to -1 on creation and assume they will be overriden? ie:

```dart
var mergedReturnStates = List<int>.filled(
      a.returnStates.length + b.returnStates.length,
      -1,
    );
```

Or should I keep up this traversal to force them to be defined?

Reply:
Based on the code I think it makes senses to traverse and assert, since returnStates shouldn't be null if the array walk funtions properly. That way we can fail early on bugs rather than having hidden -1. What do you think?

Reply:
Ive just looked into the Swift code and they fill the list with zeros:

```swift
var mergedReturnStates = [Int](repeating: 0, count: mergedReturnStatesLength)
```

Is it ok to do the same? Check the last commit for that change: https://github.com/antlr/antlr4/pull/3116/commits/9d4ca706eb5522fcd7bbc721abdb7edc898225cd